subsys/logging/Kconfig.misc

```
@@ -70,6 +70,15 @@ config LOG_FMT_SECTION_STRIP
 	depends on LOG_FMT_SECTION
 	depends on LINKER_DEVNULL_SUPPORT
 	imply LINKER_DEVNULL_MEMORY
+	imply LOG_FMT_STRING_VALIDATE
+
+config LOG_FMT_STRING_VALIDATE
+	bool "Validate logging strings"
+	help
+	  Logging strings cannot use %p with character pointers. They shall be
+	  casted to the pointer of other type. When feature is enabled, at
+	  compile time preprocessor detects strings with %p without casting and
+	  report error at runtime.
```

Some description improvements. What is the "other type" you are referring to?

```suggestion
	  Logging strings cannot use %p with character pointers. They shall be
	  casted to the pointer of another type. With this feature is enabled, at
	  compile time, the preprocessor detects strings with %p without casting
	  and reports an error at runtime.
```

---

Added `another type (e.g. void *)`. Basically it means that it shall be casted to anything other than `char *` or `unsigned char *` and simplest is `void *` to indicate that it is a pointer and for `%p` it does not matter what is the type.